### PR TITLE
Refs #32258 - Added documentation about status class

### PR DESCRIPTION
**Pull request by ShimShtein**

### CODE REVIEW

File: developer_docs/how_to_create_a_plugin.asciidoc [hunk 1]
```diff
@@ -1330,6 +1330,19 @@ class RandomStatus < HostStatus::Status
...
 ----
 
+The status class _must_ implement the followig methods:
+
+ * `to_label`: this method will be called to determine the string that will be used while displaying the status value.
+ * `self.status_name`: this method will be called to determine what label to display for the status.
+
+It _can_ also implement the following methods according to the specific needs:
+
+ * `to_global`: This method will be used to determine global status according to this specific one. The mechanism here is "voting" - to_global is called for each status and the bigger value from https://github.com/theforeman/foreman/blob/ceb276fbe96b97770f5d292b1eadca0205a34a0a/app/models/host_status/global.rb#L3[the list] would be taken. The default is `HostStatus::Global::OK`.
```
Comment (ekohl) [inline developer_docs/how_to_create_a_plugin.asciidoc]: Instead of `bigger value`, souldn't it be `highest value`?

**Review by ekohl — DISMISSED** [review]

File: developer_docs/how_to_create_a_plugin.asciidoc (hunk 1 above)
Comment (ShimShtein) [reply developer_docs/how_to_create_a_plugin.asciidoc]: Right! Fixed.

**Review by ezr-ondrej — APPROVED** [review]
Thanks @ShimShtein !

**Review by ekohl — APPROVED** [review]
No need to wait for a complete build on a docs change.

**Comment by ShimShtein** [comment]
@ekohl, can we exclude docs folder from tests trigger? IMHO it would be nice.

**Comment by ekohl** [comment]
I'm not sure if we can specify that. So far it hasn't been important enough to optimize for it and we have people who can merge while tests are still running.

**Pull request merged**